### PR TITLE
feat: display the app list search results based on their relevancy

### PR DESCRIPTION
**Pull request by Malyue**
#### What this PR does / why we need it:
display the app list search results based on their relevancy(sorted based on their length of display_name)

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=562550&iterationID=12783&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    display the app list search results based on their relevancy          |
| 🇨🇳 中文    |    新增应用模糊查询时按匹配度排序（根据查询的字段长度）          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/core/legacy/dao/application.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func (client *DBClient) GetApplicationsByIDs(orgID *int64, projectID *int64, app
 		db = db.Order(fmt.Sprintf("%s", request.OrderBy))
 	}
 	if request.Query != "" {
-		db = db.Where("name LIKE ? OR display_name LIKE ?", strutil.Concat("%", request.Query, "%"), strutil.Concat("%", request.Query, "%"))
+		db = db.Where("name LIKE ? OR display_name LIKE ?", strutil.Concat("%", request.Query, "%"), strutil.Concat("%", request.Query, "%")).Order(gorm.Expr("CHAR_LENGTH(display_name)")).Order(gorm.Expr("CHAR_LENGTH(name)"))
```
Comment (sfwn) [inline internal/core/legacy/dao/application.go:134]: Test `full text` index.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/core/legacy/dao/application.go (hunk 1 above)
Comment (Malyue) [reply internal/core/legacy/dao/application.go]: Full-text index is a little cumbersome compared to simple matching and requires additional memory consumption. This field is quite short, making a direct `like` search and sort by their char_length and the index of the first occurrence more suitable.

Comment (Malyue) [reply internal/core/legacy/dao/application.go]: However, use `gorm.Expr()` in mysql to sort the records can't use sqlite3 to write unit-test to test it, the function be used is only in mysql.

File: internal/core/legacy/dao/application.go [hunk 2]
```diff
@@ -131,7 +131,9 @@ func (client *DBClient) GetApplicationsByIDs(orgID *int64, projectID *int64, app
 		db = db.Order(fmt.Sprintf("%s", request.OrderBy))
 	}
 	if request.Query != "" {
-		db = db.Where("name LIKE ? OR display_name LIKE ?", strutil.Concat("%", request.Query, "%"), strutil.Concat("%", request.Query, "%"))
+		db = db.Where("name LIKE ? OR display_name LIKE ?", strutil.Concat("%", request.Query, "%"), strutil.Concat("%", request.Query, "%")).
+			Order(gorm.Expr("CHAR_LENGTH(display_name)")).
```
Comment (sfwn) [inline internal/core/legacy/dao/application.go:135]: Please describe the compatibility of this expression in SQLite.

Comment (Malyue) [reply internal/core/legacy/dao/application.go:138]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**